drivers/unix/socket_helpers.h

```diff
@@ -3,10 +3,6 @@
 
 #include <string.h>
 
-#ifndef IPV6_V6ONLY
```

Comment:
I've confirmed that on mingw-w64 >= 3.x `<ws2tcpip.h>` includes `<ws2ipdef.h>` which defines `IPV6_V6ONLY`.

But indeed on mingw-w64 2.x (what we have on Travis) the latter does not define `IPV6_V6ONLY`: https://sourceforge.net/p/mingw-w64/mingw-w64/ci/v2.x/tree/mingw-w64-headers/include/ws2ipdef.h

I guess you need to keep the workaround, and just add a comment saying that it's for compatibility with mingw-w64 2.x, so that we can remove it in a year or two.